### PR TITLE
Development

### PR DESCRIPTION
**Pull request by BalzaniEdoardo**
### Basis
- **Model identifiability**: new property, a boolean flag, will determine if the user wants to impose an identifiablity constraint to the feature matrix. If the flag is set to true, at feature creation, it will check if [1, X] is full rank, if not, it will drop columns until the rank of [1,X] will be full. Additionally, X is mean centered, this is a standard practice for B-spline regression in GAMs. Indentifiability is also needed for modelling categorical variables, for which model like "z ~ 1 + x + y + x*y" may not be identifiable. 

### GLM

- **update** and **initialize_solver** methods. These methods are useful for batching algorithms. The former runs a single optimization step, the latter, initialize the state and the parameters.

- **validate**: validate is moved into "initialize_solver", so that it is called in at the first initialization step even if you batch, but not in the update step, because you don't want to repeat validation and slow down your loop.

### Regularizer

- **inistantiate_solver**: instantiate solver will now return 3 callables, the run, update and intialize method of the solver.

### CODE REVIEW

**Review by billbrod — COMMENTED** [review]
I really like the changes to GLM with the init/update/run abstraction!

- I think, in general, the `batch_glm` API guide and the calcium imaging exmaple need more work (though #159 will help), especially more text explaining what's going on. I think it'll take a bit, so I'm inclined to add a warning at the top saying "this is under development", and work on it in a separate PR.
- There now appear to be a bunch of hidden files in `docs/api_guide` (which just contain a single line), what's going on with them? 
- I think we need more explanations / examples about what "identifiable" means for the basis. right now, there's a pretty good explanation buried in a hidden method of an abstract class, so users won't see it
- (As a note, for keeping actions up-to-date, I recommend [action-updater](https://vsoch.github.io/action-updater/index.html) if you haven't seen it before)

File: docs/api_guide/plot_05_batch_glm.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
+"""
+# Batching example
+
+Here we demonstrate how to setup and run a stochastic gradient descent in `nemos`
```
[the remark this answers is not in the document]
Comment (billbrod) [reply docs/api_guide/plot_05_batch_glm.py:4]: #161

File: docs/api_guide/plot_05_batch_glm.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+by batching and using the `update` method of the model class.
+
+"""
+# %%
+# ## Import
```
[the remark this answers is not in the document]
Comment (BalzaniEdoardo) [reply docs/api_guide/plot_05_batch_glm.py:9]: removed

**Review by BalzaniEdoardo — COMMENTED** [review]
thanks Billy, I'll push a few commits and we should be good

**Review by billbrod — APPROVED** [review]

**Pull request merged**